### PR TITLE
use a single shared ssh connection for proxying individual tcp connec…

### LINKED ISSUE #1181: kubectl proxy fails silently to tunnel connections when no ssh key is available in the ssh agent
**What happened**: I used `kubeone proxy` without having ssh keys in my ssh-agent and kubeone did not communicate that there is a problem. When I tried to contact the kubernetes api through that proxy kubectl would just hang and then print `Unable to connect to the server: Service Unavailable` and kubeone did not print any error in this stage.

**What is the expected behavior**: Fail with a desriptive error message, that tells the user that the SSH connection could not authenticate properly

**How to reproduce the issue**: Unload keys from ssh-agent, do not have a static path configured in kubeone.yaml and repeat what I stated in `What happened`

**Anything else we need to know?**

**Information about the environment**:
KubeOne version (`kubeone version`):  1.0.5
Operating system: 
Provider you're deploying cluster on:
Operating system you're deploying on:

### PR DESCRIPTION
**Pull request by erdii**
…tions

this also has the advantage of failing early when ssh authentication fails

**What this PR does / why we need it**: 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:  

Fixes #1181

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Use a single shared ssh connection for proxying individual tcp connections
```

### CODE REVIEW

File: pkg/cmd/proxy.go [hunk 1]
```diff
@@ -90,7 +98,9 @@ func setupProxyTunnel(opts *proxyOpts) error {
 
 	fmt.Println("SSH tunnel started, please open another terminal and setup environment")
 	fmt.Printf("export HTTPS_PROXY=http://%s\n", opts.ListenAddr)
-	return server.ListenAndServe()
+
+	err = server.ListenAndServe()
+	return err
```
Comment (kron4eg) [inline pkg/cmd/proxy.go:103]: The change here is no changing to good anything, please return back the `return server.ListenAndServe()`

**Review by kron4eg — COMMENTED** [review]
In general looks good, left one microscopic comment to fix.

File: pkg/cmd/proxy.go (hunk 1 above)
Comment (erdii) [reply pkg/cmd/proxy.go:103]: Nice catch! I somehow thought it would be needed for `defer tunn.Close()` to work. :see_no_evil:

**Review by kron4eg — APPROVED** [review]

**Pull request merged**